web/package.json

```diff
@@ -21,8 +21,10 @@
     "fix-imports": "eslint --no-eslintrc --no-inline-config --parser '@typescript-eslint/parser' --plugin 'unused-imports' --plugin 'react-hooks' --rule 'unused-imports/no-unused-imports:error' --fix \"./src/**/*.{ts,tsx}\""
   },
   "devDependencies": {
-    "@openshift-console/dynamic-plugin-sdk": "0.0.2",
-    "@openshift-console/dynamic-plugin-sdk-webpack": "0.0.2",
+    "@openshift-console/dynamic-plugin-sdk": "0.0.3",
```

Comment:
just to understand the strategy regarding deps, is it correct to say that:
- all "normal" dev dependencies (ie. not used at runtimes) are in `devDependencies`, as usual
- runtime dependencies that are already provided by the console are in `devDependencies`
- runtime dependencies that are not provided by the console are in `dependencies`

Correct?

Reply:
Exactly ! 
I would also add conflicting version dependencies in `dependencies` since we disabled `concatenateModules`
For now everything works fine in 4.10. I would try a 4.9 just to check

Reply:
Righ now we don't use the `--production` flag so my understanding is that `dependencies` and `devdependencies` are installed the same way and interchangeable.

My understanding is that anything not needed to build or run the application (tests framework, linter...) should go to `devdependencies`, while anything required either at build time or runtime should go to `dependencies`

Reply:
the `--mode=production` option is set at the end of `webpack.config.ts` so it's already using that when you run `build` command 

```ts
if (process.env.NODE_ENV === 'production') {
  config.mode = 'production';
```

> My understanding is that anything not needed to build or run the application (tests framework, linter...) should go to `devdependencies`, while anything required either at build time or runtime should go to `dependencies`

This is right for any standard project, but here we are embedded in the console using simple `import` statements. That's why they have some difficulties to catch errors btw

Check Sam Padgett [console-plugin-template](https://github.com/spadgett/console-plugin-template) for example

Reply:
> the --mode=production option is set at the end of webpack.config.ts so it's already using that when you run build command

Yes but the flag is still not used during the `npm install` step previous to the build and adding the flag there make the build fails due to missing dependencies.

Reply:
Here is the trick

```ts
  // Manage dependencies from package.json file. Replace all devDependencies 'import' by 'require'
  config.externals = NodeExternals({
    fileName: './package.json',
    includeInBundle: ['dependencies'],
    excludeFromBundle: ['devDependencies']
  });
```